src/openassetio-python/package/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -708,6 +712,493 @@ def __testGetWithRelationshipsError(
         self.assertEqual(actual_error, expected_error)
 
 
+class PagedTest(FixtureAugmentedTestCase):
+    """
+    Helpers for tests that utilize a Pager-like object
+    """
+
+    class weaklist(list):
+        """
+        Built-in `list` type does not support weakref, so create
+        this shim.
+        """
+
+        __slots__ = ("__weakref__",)
+
+    def _assert_pager_is_at_end(self, pager):
+        self.assertListEqual(pager.get(), [])
+        self.assertFalse(pager.hasNext())
+        pager.next()
+        self.assertListEqual(pager.get(), [])
+        self.assertFalse(pager.hasNext())
+        pager.next()
+        self.assertListEqual(pager.get(), [])
+        self.assertFalse(pager.hasNext())
+
+    @classmethod
+    def _concat_all_pages(cls, pager):
+        return list(cls._pager_elems_iter(pager))
+
+    @classmethod
+    def _pager_elems_iter(cls, pager):
+        for page in cls._pager_page_iter(pager):
+            for elem in page:
+                yield elem
+
+    @staticmethod
+    def _pager_page_iter(pager):
+        while page := pager.get():
```

Comment:
[question] This fails on Python 3.7 (walrus unsupported).  Do we still need to support 3.7?

Reply:
Sadly so - many are still on CY21